workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -352,6 +355,62 @@ def _restore_deployment_envs(self, postgres):
 
         self._log_final_information(deps_with_failed_plugins)
 
+    def _restore_inter_deployment_dependencies(self):
+        if (self._snapshot_version < V_4_6_0 or
+                self._snapshot_version > V_5_0_5):
+            return
+        ctx.logger.info('Restoring inter deployment dependencies')
+        update_service_composition = (self._snapshot_version == V_5_0_5)
+
+        deployment_contexts = utils.get_dep_contexts(self._snapshot_version)
+        deployments_queue = queue.Queue()
+        failed_deployments_queue = queue.Queue()
+        for tenant_name, deployments in deployment_contexts:
+            for dep_id in deployments:
+                deployments_queue.put((tenant_name, dep_id))
+
+        wf_context = current_workflow_ctx.get_ctx()
+        context_params = current_workflow_ctx.get_parameters()
+
+        def _create_inter_deployment_dependencies():
+            while True:
+                try:
+                    tenant, deployment_id = deployments_queue.get_nowait()
+                except queue.Empty:
+                    break
+
+                with current_workflow_ctx.push(wf_context, context_params):
+                    try:
+                        tenant_client = get_rest_client(tenant=tenant)
+                        tenant_client.inter_deployment_dependencies.restore(
+                            deployment_id, update_service_composition)
+                    except RuntimeError as err:
+                        failed_deployments_queue.put((deployment_id, tenant))
+                        ctx.logger.info('Failed creating inter deployment '
+                                        'dependencies for deployment {0} from'
+                                        ' tenant {1}. {2}'
+                                        .format(deployment_id, tenant, err))
+
+        threads = list()
```

Comment:
```suggestion
        threads = []
```
`[]` is just shorter to write